bson/unmarshaling_cases_test.go

```diff
@@ -78,3 +78,94 @@ var unmarshalingTestCases = []unmarshalingTestCase{
 		docToBytes(D{{"fooBar", int32(10)}}),
 	},
 }
+
+var unmarshalingExtTestCases = []unmarshalingTestCase{
+	{
+		name: "Small struct",
+		reg:  DefaultRegistry,
```

Comment:
Looks like these all use the same `reg` so might not need to be a test case variable.

Reply:
Agreed, will update to use a different struct for these tests that doesn't include a registry.

Reply:
Moved test cases into test function and used a test-specific struct type to remove the unused registry variable.